### DISCUSSION TITLE
default value as output with pipe?

### DISCUSSION THREAD

**Discussion by ha1fstack** [open-circle/valibot#657] [Q&A] [answered]
https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6BjCAO1XgGUmALAUxDI4AXkwA6CGBjA2ZABQYxYYGD4KxHKMFYBzOQEoANOJhRk7bNBBywyKKj4BJVjH1GiABkL6GzNhzgoPlQAV1J4UUVUZGw+AAU7BzlufkFjENYAEz5sbT5Mn0YWdghSPjFSCD0g0PDCoA

```ts
import * as v from 'valibot';

const Schema = v.optional(v.pipe(v.string(), v.transform(parseInt)), '0');

const result = v.safeParse(Schema, undefined);
console.log(result);
```

Current 'default value' implementation uses the provided default value as 'input' to the wrapped schema. However I want to provide default value as 'output' of wrapped schema so it skips the pipe and I can provide the desired default output value directly like so:

```ts
const Schema = v.optional(v.pipe(v.string(), v.transform(parseInt)), 0); // number 0 instead of string '0'
```

How should I implement this?

**Comment by fabian-hiller** [comment, marked as answer]
I would rewrite your schema. See [this playground](https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6BjCAO1XgGUmALAUxDI4AXkwA6AK6tgbABQBtenHFhgYPrIxiOUYKwDmsgJQAacTCjJ22aCFlhkUVHwCSrGEdNLxEMDBmsZJpirBIgFHxQxmYADF4AukYMzGwccFB8qBKk8KJaqMjYfAAKjs6y3PyCZlIAJnzYeny1SYws7BCkfGKkEIYZWTmtQA).

```ts
import * as v from 'valibot';

const Schema = v.union([
  v.pipe(v.string(), v.transform(parseInt)),
  v.optional(v.number(), 0),
]);
```

Another option is to use the [`fallback`](https://valibot.dev/api/fallback/) method instead of the `optional` schema, but the behavior is different. See [this playground](https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6BjCAO1XgGUmALAUxDI4AXkwA6bGVIVkTANYAKDGLDAwfJWI5RgrAOYKAlABpxMKMnbZoIBWGRRUfAJKsYhk3AAMhhszYccFB8qACupPCiyqjI2HwACg5OCtz8gqahrAAmfNi6fFm+jCzsEKR8YqQQBsFhEUVAA).

```ts
import * as v from 'valibot';

const Schema = v.fallback(v.pipe(v.string(), v.transform(parseInt)), 0);
```

**Reply by ha1fstack** [reply]
Thank you for the detailed answer! The latter implementation suits my case since I only want number input for the field.
